stdlib/LinearAlgebra/src/triangular.jl

```diff
@@ -50,6 +50,20 @@ for t in (:LowerTriangular, :UnitLowerTriangular, :UpperTriangular,
     end
 end
 
+for t in (:LowerTriangular, :UnitLowerTriangular, :UpperTriangular,
+          :UnitUpperTriangular)
+    @eval begin
+        adjtypes = Dict(
+            LowerTriangular => UpperTriangular,
+            UpperTriangular => LowerTriangular,
+            UnitLowerTriangular => UnitUpperTriangular,
+            UnitUpperTriangular => UnitLowerTriangular,
+        )
+        similar(A::Adjoint{TI,TV}, ::Type{T}) where{T,TI,TV<:$t} = adjtypes[$t](similar(parent(parent(A)), T))
+        similar(A::Transpose{TI,TV}, ::Type{T}) where{T,TI,TV<:$t} = adjtypes[$t](similar(parent(parent(A)), T))
+    end
+end
+
```

Comment:
If I'm not mistaken, then writing this out explicitly yields 8 lines of clear code, as opposed to, well, what you have. 😛

Reply:
That's correct, I did not think about this...